### ISSUE TITLE
Non-admin users should not be able to cancel other admin queries or sessions from the console 

### ISSUE THREAD

**Issue by kevin-v-ngo** [cockroachdb/cockroach#75758]
As described in our [docs](https://www.cockroachlabs.com/docs/v21.2/create-role#role-options), non-admin users should not be able to cancel other admin queries or sessions.

Repro:
1. Run statements from an admin user
2. Create a non-admin user with VIEWACTIVITYREDACTED role option
3. Go to the Sessions page as the non-admin user and click Terminate session on the admin's session

From the steps above, CANCELQUERY role option has never been given to the user. And even with the CANCELQUERY role option, this behavior should not be supported. 

gz#11226

**Comment by ajwerner** [comment]
The code around this is pretty clear, it'll be interesting to see what's up here:

https://github.com/cockroachdb/cockroach/blob/73b681967bf0a0da60149d50b111f25000954995/pkg/server/status.go#L236-L296

**Comment by kevin-v-ngo** [comment]
Thanks Andrew. FYI @maryliag

**Comment by matthewtodd** [comment]
@kevin-v-ngo, just wanted to clarify the scope here. IIUC, here's what we're covering:

- admin: can cancel all query and sessions
- non-admin with `CANCELQUERY`: can cancel their own session and query and other non-admin sessions and queries
- non-admin without `CANCELQUERY`: can cancel their own session and queries and nothing else

Is that right?

**Comment by matthewtodd** [comment]
Hmm, perhaps I'm missing something, but I'm unable to reproduce this locally. I've created 3 users, alice (admin), bob (`VIEWACTIVITYREDACTED`, `CANCELQUERY`), and carol (`VIEWACTIVITYREDACTED`). This table summarizes the findings, where the rows are actors and the columns are the owners of the target statements to be canceled. As expected, alice can cancel everyone's statements, bob can only cancel non-admin statements, and carol can only cancel her own.

||root|alice|bob|carol|
|--|--|--|--|--|
|alice|✅|✅|✅|✅|
|bob|❌|❌|✅|✅|
|carol|❌|❌|❌|✅|

I'm wondering if something like the `--insecure` flag might be affecting the behavior observed elsewhere.

**Comment by lancel66** [comment]
> Hmm, perhaps I'm missing something, but I'm unable to reproduce this locally. I've created 3 users, alice (admin), bob (`VIEWACTIVITYREDACTED`, `CANCELQUERY`), and carol (`VIEWACTIVITYREDACTED`). This table summarizes the findings, where the rows are actors and the columns are the owners of the target statements to be canceled. As expected, alice can cancel everyone's statements, bob can only cancel non-admin statements, and carol can only cancel her own.
> 
> root	alice	bob	carol
> alice	✅	✅	✅	✅
> bob	❌	❌	✅	✅
> carol	❌	❌	❌	✅
> I'm wondering if something like the `--insecure` flag might be affecting the behavior observed elsewhere.

Not sure if this is relevant, but the issue was experienced on a nightly build: `V21.2.4-153-GBA879C1D54`

**Comment by matthewtodd** [comment]
Thanks, @lancel66, I'll pull down that build and see if it makes a difference.

And for reference, here's how I'm starting the database:

````bash
#!/bin/bash

set -euo pipefail

COCKROACH=${GOPATH}/src/github.com/cockroachdb/cockroach/cockroach
ROOT=$(cd $(dirname ${BASH_SOURCE[0]}); pwd)
export COCKROACH_CA_KEY=${ROOT}/certs/ca.key
export COCKROACH_CERTS_DIR=${ROOT}/certs

rm -rf ${COCKROACH_CERTS_DIR}

${COCKROACH} cert create-ca \
  --ca-key=${COCKROACH_CA_KEY} \
  --certs-dir=${COCKROACH_CERTS_DIR} \
  --overwrite

${COCKROACH} cert create-node localhost \
  --ca-key=${COCKROACH_CA_KEY} \
  --certs-dir=${COCKROACH_CERTS_DIR} \
  --overwrite

${COCKROACH} start-single-node \
  --certs-dir=${COCKROACH_CERTS_DIR} \
  --cert-principal-map=localhost:node \
  --listen-addr=localhost:26257 \
  --http-addr=localhost:8080 \
  --unencrypted-localhost-http
````

And here's how I'm starting each user's sql session:

```bash
#!/bin/bash

set -euo pipefail

COCKROACH=${GOPATH}/src/github.com/cockroachdb/cockroach/cockroach
ROOT=$(cd $(dirname ${BASH_SOURCE[0]}); pwd)
export COCKROACH_CA_KEY=${ROOT}/certs/ca.key
export COCKROACH_CERTS_DIR=${ROOT}/certs

USERNAME=${1:-root}

${COCKROACH} cert create-client ${USERNAME} \
  --ca-key=${COCKROACH_CA_KEY} \
  --certs-dir=${COCKROACH_CERTS_DIR} \
  --overwrite

${COCKROACH} sql \
  --url "postgresql://${USERNAME}@localhost:26257?ssl-mode=verify-full&sslrootcert=${COCKROACH_CERTS_DIR}/ca.crt&sslcert=${COCKROACH_CERTS_DIR}/client.${USERNAME}.crt&sslkey=${COCKROACH_CERTS_DIR}/client.${USERNAME}.key"
```
</details>

**Comment by matthewtodd** [comment]
I've just tried locally with `v21.2.4-153-gba879c1d54` and I'm still seeing the expected, not the buggy, behavior.

**Comment by matthewtodd** [comment]
Closing the loop here, this afternoon's hypothesis of the request forwarding not carrying the proper authentication information turned out to be correct. There's a fix building now in #75814, backported to 21.2 in #75829, which should land in tonight's 21.2 build.
